docs/proposals/volumes.md

```diff
@@ -0,0 +1,431 @@
+    2.  Volumes based on network block devices: AWS EBS, iSCSI, RBD, etc
+2.  Shared storage: NFS, Glusterfs, Cephfs, etc.  For these volumes, the
+    ownership is determined by the configuration of the shared storage system.
+    1.  `hostPath` is a special case of shared storage
+
+The `EmptyDir` volume was recently modified to create the volume directory with `0777` permissions
+from `0750` to support basic usability of that volume as a non-root UID.
+
+### Docker
+
+Docker recently added supplemental group support.  This adds the ability to specify additional
+groups that a container should be part of, and will be released with Docker 1.8.
+
+There is a [proposal](https://github.com/docker/docker/pull/14632) to add a bind-mount flag to tell
+Docker to change the ownership of a volume to the effective UID and GID of a container, but this has
+not yet been accepted.
+
+### Rocket
+
+Rocket
+[image manifests](https://github.com/appc/spec/blob/master/spec/aci.md#image-manifest-schema) can
+specify users and groups, similarly to how a Docker image can.  A Rocket
+[pod manifest](https://github.com/appc/spec/blob/master/spec/pods.md#pod-manifest-schema) can also
+override the default user and group specified by the image manifest.
+
+Rocket does not currently support supplemental groups or changing the owning UID or
+group of a volume, but it has been [requested](https://github.com/coreos/rkt/issues/1309).
+
+## Use Cases
+
+1.  As a user, I want the system to set ownership and permissions on volumes correctly to enable
+the following scenarios:
+    1.  All containers running as root
+    4.  All containers running as the same non-root user
+    5.  Multiple containers running as a mix of root and non-root users
+
+### All containers running as root
+
+For volumes that only need to be used by root, no action needs to be taken to change ownership or
+permissions, but setting the ownership based on the supplemental group shared by all containers in a
+pod will also work.  For situations where read-only access to a shared volume is required from one
+or more containers, the `VolumeMount`s in those containers should have the `readOnly` field set.
+
+### All containers running as a single non-root user
+
+In use cases whether a volume is used by a single non-root UID the volume ownership and permissions
+should be set to enable read/write access.
+
+Currently, a non-root UID will not have permissions to write to any but an `EmptyDir` volume.
+Today, users that need this case to work can:
+
+1.  Grant the container the necessary capabilities to `chown` and `chmod` the volume:
+    - `CAP_FOWNER`
+    - `CAP_CHOWN`
+    - `CAP_DAC_OVERRIDE`
+2.  Run a wrapper script that runs `chown` and `chmod` commands to set the desired ownership and
+    permissions on the volume before starting their main process
+
+This workaround has significant drawbacks:
+
+1.  It grants powerful kernel capabilities to the code in the image
```

Comment:
`... and thus is not secure, defeating the reason containers are run as non-root users`